ledger/src/blockstore_processor.rs

```diff
@@ -945,17 +937,24 @@ fn load_frozen_forks(
                 cache_block_time_sender,
                 None,
                 timing,
-            )
-            .is_err()
-            {
+            );
+            // Insert even dead banks into the BankForks so that cleanup
+            // of account state created on Bank creation via `Bank::new_from_parent()`
+            // will occur when the bank is dropped. Otherwise, this state is not
+            // cleaned
+            all_banks.insert(bank.slot(), bank.clone());
+
+            if let Err(e) = process_result {
+                warn!("processing single slot {} error {:?}", slot, e);
+                initial_forks.insert(bank.slot(), bank.clone());
+                all_banks.insert(bank.slot(), bank.clone());
                 continue;
             }
             txs += progress.num_txs;
 
             // Block must be frozen by this point, otherwise `process_single_slot` would
             // have errored above
```

Comment:
nits: `// Block must be frozen by this point because any errors should have been bailed out above.`